CARDS-2383: Create importable spreadsheet for the new YVM questionnaire

marta- opened the pull request:
DRAFT until #1650 is merged (adding support for custom summary screen message).

Added the questionnaire and an initial survey configuration, as well as an extremely basic clinic configuration (just enough to facilitate testing).

To test:
* build this branch
* start with `-P  prems`
* create a visit for the Your Voice Matters (PMH) "clinic"
* obtain the auth token
* as patient, start filling out the YVM survey
  * check that the message on the start screen corresponds to the intro from the [specs PDF](https://phenotips.atlassian.net/browse/CARDS-2383)
  * check that the overall survey is according to [specs PDF](https://phenotips.atlassian.net/browse/CARDS-2383)
  * check that the message on the summary screen corresponds to the ending from the [specs PDF](https://phenotips.atlassian.net/browse/CARDS-2383)

marta- committed:
CARDS-2383: Create importable spreadsheet for the new YVM questionnaire

sashaandjic commented:
2 issues that I was not able to address in the spreadsheet:
1.  The 2 "No.." options should act as "None of the above"
![image](https://github.com/data-team-uhn/cards/assets/23243244/21b48953-7f6a-431e-b634-4082215657d7)
In the designer I was abe to repace "N/A' and "none of the above" options, however one of them was displayed at the top, and it should not:
![image](https://github.com/data-team-uhn/cards/assets/23243244/7eb945c5-7f08-405c-966a-4cf8832130fe)
![image](https://github.com/data-team-uhn/cards/assets/23243244/3418b73e-efe4-48be-8739-0ddadc807058)


2. The "I did not have any of these problems" should act as "none of the above"
![image](https://github.com/data-team-uhn/cards/assets/23243244/7cb217d5-852d-414c-bc25-1caa992a55a2)
I was able to make this change in the designer and it worked as expected
![image](https://github.com/data-team-uhn/cards/assets/23243244/1a088d4f-139d-4aad-96a4-a080e555f2e0)

sashaandjic commented:
![image](https://github.com/data-team-uhn/cards/assets/23243244/610fcf3c-6597-4791-afa1-8312b9aa31d0)

https://github.com/data-team-uhn/cards/assets/23243244/d4a36b6f-bc0b-40dd-9573-f073a5e229e3

After I hit "REVIEW" button for the first time, an error msg showed up for a second.

marta- committed:
CARDS-2383: Create importable spreadsheet for the new YVM questionnaire

Updated the text on the summary screen

marta- committed:
CARDS-2383: Create importable spreadsheet for the new YVM questionnaire

* Added hidden reference questions (hospital, department)
* Fixed whitespace in texts (removed unnecessary leading spaces and interhan new lines)
* Switched most option values from text to long
* Fixed section 6b condition
* Split long page (originally page 3) into 3 shorter pages

marta- committed:
CARDS-2383: Create importable spreadsheet for the new YVM questionnaire

Marked na/none of the above options in multiselect questions

marta- committed:
CARDS-2384: Configure the YVM clinic
Added an initial clinic definition to facilitate testing the survey. The configuration is subject to change.

marta- commented:
> ## yvm_6b condition did not get imported

Fixed in the latest version of the spreadsheet and  in commit 12f384a.

> **1.** Two options at the bottom, should each act as `None of the above` option 
> **2.** The option "I did not have any of these problems" should act as `None of the above` 

Done in 7d7d927.

marta- commented:
> after I hit the "REVIEW" button for the first time, an error msg showed up for a second.

I'm still unable to replicate this. If you manage to catch the actual error please post it here.

sashaandjic commented:
Everything looks perfect except for one problem: In order to get YVM survey created my visit date needed to be 30 days ago. However, the generated token would not open the survey as it seems to be expired. Once I moved the visit date to 7 days ago, the new token worked.

sashaandjic commented:
> > after I hit the "REVIEW" button for the first time, an error msg showed up for a second.
> 
> I'm still unable to replicate this. If you manage to catch the actual error please post it here.

I can no longer reproduce it either.

sashaandjic reviewed (approved):
tested
![image](https://github.com/data-team-uhn/cards/assets/23243244/0b28efaf-6e0f-473a-9ab0-3d52fe11b70b)
![image](https://github.com/data-team-uhn/cards/assets/23243244/5fda37aa-739c-4d9d-9c75-e30fb95abbb4)
![image](https://github.com/data-team-uhn/cards/assets/23243244/b77f0d05-d372-4602-958f-38ad1f10919a)
![image](https://github.com/data-team-uhn/cards/assets/23243244/6b05ee69-b4a4-4def-b4b8-27ca62106d35)
![image](https://github.com/data-team-uhn/cards/assets/23243244/a6c204ad-19ee-4b1b-9a60-77f6a6595349)
![image](https://github.com/data-team-uhn/cards/assets/23243244/812fe077-008c-43b5-8a9c-e45662c3c62c)
![image](https://github.com/data-team-uhn/cards/assets/23243244/ad3e92cf-1858-4a85-a734-fb968523f94a)

sashaandjic commented:
There is an error in the spreadsheet. I forgot to put `MaxAnswers=0` for `yvm_department`.
As a consequence, if visit info form has more than 1 privoder, the YVM form will be marked as `Invalid`.
![image](https://github.com/data-team-uhn/cards/assets/23243244/fd22358c-e2b1-4653-a819-cf07422d5328)

Once I modified `MaxAnswers` from `1` to `0`, the problem got resolved.
-----
![image](https://github.com/data-team-uhn/cards/assets/23243244/5d9da394-deeb-4f5c-bcac-b87ac66bfdf3)

sashaandjic commented:
QUESTION: Should we extend the Survey expiration for YVM?

marta- committed:
CARDS-2383: Create importable spreadsheet for the new YVM questionnaire

Fixed maxAnswers of reference question to match its source.

marta- commented:
> There is an error in the spreadsheet. I forgot to put `MaxAnswers=0` for `yvm_department`.

Updated on f91e1dc

marta- commented:
> QUESTION: Should we extend the Survey expiration for YVM? When do we stop sending YVM surveys? We send them monthly if there was no response, and every 3 months if patient responded.

We will obtain the final schedule from the client soon. For now, 30 days from the visit is reasonable, as the invitation emails will be sent at most 48 hours after the visit with reminders probably 2 days apart.

marta- committed:
CARDS-2383: Create importable spreadsheet for the new YVM questionnaire

* Fixed typo in new reference questions' node names
* Fixed reference of answers copied over to the top level of the YVM form json

sashaandjic reviewed (approved):
tested - all good

The "Thank you" page for other prems surveys vs `YVM` survey:
![image](https://github.com/data-team-uhn/cards/assets/23243244/f01681eb-4b40-4a6b-8ed2-3b810fc955cf)

sdumitriu commented on prems-resources/clinical-data/pom.xml:
```diff
@@ -50,6 +50,7 @@
               SLING-INF/content/Survey/IPIC.xml;path:=/Survey/IPIC;overwrite:=true,
               SLING-INF/content/Survey/EDIPIC.xml;path:=/Survey/EDIPIC;overwrite:=true,
               SLING-INF/content/Survey/Rehab.xml;path:=/Survey/Rehab;overwrite:=true,
+              SLING-INF/content/Survey/YVM.xml;path:=/Survey/Rehab;overwrite:=true,
```
```suggestion
              SLING-INF/content/Survey/YVM.xml;path:=/Survey/YVM;overwrite:=true,
```

sdumitriu replied:
Actually, do we expect any other YVM questionnaires? In the follow up PR I see that it's sometimes named "PMH-YVM".

marta- replied:
Copy-paste artifact, thanks for noticing. No other questionnaires expectected for yvm.

marta- replied:
Fixed

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/YVM.xml:
```diff
@@ -0,0 +1,109 @@
+    <property>
+        <name>intro</name>
+        <value>
+**Your Voice Matters** is a set of questions that gives you the chance to share your cancer care
```
I would use italic instead of bold here.

marta- replied:
I followed the template that was provided. I will bring it up at the next meeting, but it shouldn't prevent the merge. Other adjustments are expected anyway.

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/YVM.xml:
```diff
@@ -0,0 +1,109 @@
+    <property>
+        <name>ending</name>
+        <value>
+Thank you for completing the Your Voice Matters survey.
```
I would make YVM italic here too.

marta- replied:
Same response as above.

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/YVM.xml:
```diff
@@ -0,0 +1,109 @@
+    {
+      "key": "yvm_14",
+      "label": "Overall experience",
+      "link" : "string"
```
??

This has been copied into all the clinics, but I don't think it makes sense. It can be completely removed, it doesn't need a format or a link at all.

In the future, we could try to link to the stats view, but that's not possible right now.

marta- replied:
Please clarify, you want to remove te link field?

sdumitriu replied:
Yes.

marta- replied:
Removed

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/YVM.xml:
```diff
@@ -0,0 +1,109 @@
+the cancer care experience.
+
+If you want to provide detailed feedback about your experience directly to the hospital, please
+contact [patientexperience@uhn.ca](mailto:patientexperience@uhn.ca).
```
I would add this, like in the YE surveys:

> Your responses have been received. You can now close this browser window.

And thinking about it, why not "You can now close this page"?

sashaandjic replied:
I was also thinking about this, but they requested this exact text.
@marta- would it be ok to add it?

marta- replied:
I'll add it for now and explain it's basically part of our portal's instructions. I will get confirmation whether we can keep it at the next call

marta- replied:
Done

marta- committed:
CARDS-2383: Create importable spreadsheet for the new YVM questionnaire

Addressing review comments:
* Fixed copy-paste artifact
* Improved summary screen text
* Removed unnecessary field for clinic dashboard column

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/YVM.xml:
```diff
@@ -50,6 +50,8 @@ the cancer care experience.
 If you want to provide detailed feedback about your experience directly to the hospital, please
 contact [patientexperience@uhn.ca](mailto:patientexperience@uhn.ca).
+
+Your answers have been received. You can now close this page.
```
For YE it's bold, should it be bold here too?

sashaandjic replied:
Here are the screenshots for YE and YVM Thank you page:
![image](https://github.com/data-team-uhn/cards/assets/23243244/92310921-f685-4ead-9852-adec7102ebd4)

marta- replied:
Updated in the CARDS-2384 branch (PR #1651) in commit ef64df9 .

sashaandjic reviewed (approved):
tested

sashaandjic commented:
YE and YVM Welcome pages:
![image](https://github.com/data-team-uhn/cards/assets/23243244/2aa166bf-c572-4cb5-bb80-11482b444227)
